lib/boom_notifier/mail_notifier.ex

```diff
@@ -32,31 +32,31 @@ defmodule BoomNotifier.MailNotifier do
   @type options :: [option]
 
   @impl BoomNotifier.Notifier
-  def validate!(options) do
-    with _mailer <- Keyword.fetch!(options, :mailer),
-         _from <- Keyword.fetch!(options, :from),
-         _to <- Keyword.fetch!(options, :to),
-         _subject <- Keyword.fetch!(options, :subject) do
-      nil
+  def validate_config(options) do
+    with {:mailer, {:ok, _mailer}} <- {:mailer, Keyword.fetch(options, :mailer)},
+         {:from, {:ok, _from}} <- {:from, Keyword.fetch(options, :from)},
+         {:to, {:ok, _to}} <- {:to, Keyword.fetch(options, :to)},
+         {:subject, {:ok, _subject}} <- {:subject, Keyword.fetch(options, :subject)} do
+      :ok
+    else
+      {:mailer, :error} -> {:error, "Mailer is missing"}
```

Comment:
I've thought about this a little more and I think I like @jmbejar approach about not using `with` for this. Maybe something like:

```elixir
missing_keys = Enum.filter([:mailer, :from, :to, :subject], fn key ->
  !Keyword.has_key?(options, key)
end)

case missing_keys do
  [] -> :ok
  [missing_key] -> {:error, "#{missing_key} parameter is missing"}
  _ ->  {:error, "The following parameters are missing: #{inspect(missing_keys)}"}
```

Let me know whether this makes sense to you